api/src/main/java/uk/gov/dhsc/htbhf/dwp/factory/EligibilityResponseFactory.java

```diff
@@ -1,43 +1,71 @@
 package uk.gov.dhsc.htbhf.dwp.factory;
 
+import org.springframework.stereotype.Component;
 import uk.gov.dhsc.htbhf.dwp.entity.Child;
 import uk.gov.dhsc.htbhf.dwp.entity.legacy.LegacyHousehold;
 import uk.gov.dhsc.htbhf.dwp.entity.uc.UCHousehold;
+import uk.gov.dhsc.htbhf.dwp.model.ChildDTO;
 import uk.gov.dhsc.htbhf.dwp.model.EligibilityResponse;
 import uk.gov.dhsc.htbhf.eligibility.model.EligibilityStatus;
 
 import java.time.LocalDate;
+import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
+/**
+ * Generates an {@link EligibilityResponse} for a given household ({@link LegacyHousehold} or {@link UCHousehold}) and {@link EligibilityStatus}.
+ */
+@Component
 public class EligibilityResponseFactory {
 
-    public static EligibilityResponse createEligibilityResponse(LegacyHousehold household, EligibilityStatus eligibilityStatus) {
+    public EligibilityResponse createEligibilityResponse(LegacyHousehold household, EligibilityStatus eligibilityStatus) {
         return EligibilityResponse.builder()
                 .numberOfChildrenUnderOne(getNumberOfChildrenUnderOne(household.getChildren()))
                 .numberOfChildrenUnderFour(getNumberOfChildrenUnderFour(household.getChildren()))
                 .householdIdentifier(household.getHouseholdIdentifier())
+                .children(getChildrenUnderFour(household))
                 .eligibilityStatus(eligibilityStatus)
                 .build();
     }
 
-    public static EligibilityResponse createEligibilityResponse(UCHousehold household, EligibilityStatus eligibilityStatus) {
+    public EligibilityResponse createEligibilityResponse(UCHousehold household, EligibilityStatus eligibilityStatus) {
         return EligibilityResponse.builder()
                 .householdIdentifier(household.getHouseholdIdentifier())
                 .numberOfChildrenUnderFour(getNumberOfChildrenUnderFour(household.getChildren()))
                 .numberOfChildrenUnderOne(getNumberOfChildrenUnderOne(household.getChildren()))
+                .children(getChildrenUnderFour(household))
                 .eligibilityStatus(eligibilityStatus)
                 .build();
     }
 
-    private static Integer getNumberOfChildrenUnderOne(Set<? extends Child> children) {
+    private List<ChildDTO> getChildrenUnderFour(UCHousehold household) {
+        return household.getChildren().stream()
+                .filter(child -> isUnderFour(child.getDateOfBirth()))
+                .map(child -> ChildDTO.builder().dateOfBirth(child.getDateOfBirth()).build())
+                .collect(Collectors.toList());
+    }
+
+    private List<ChildDTO> getChildrenUnderFour(LegacyHousehold household) {
```

Comment:
Can we have a shared interface for `UCHousehold` and `LegacyHousehold` so that we can remove this otherwise duplicate method?

Reply:
Done